apm-agent-core/src/main/java/co/elastic/apm/agent/impl/transaction/AbstractSpan.java

```diff
@@ -164,7 +164,22 @@ public StringBuilder getNameForSerialization() {
      */
     @Nullable
     public StringBuilder getAndOverrideName(int namePriority) {
-        if (namePriority >= this.namePriority) {
+        return getAndOverrideName(namePriority, true);
+    }
+
+    /**
+     * Resets and returns the name {@link StringBuilder} if one of the following applies:
+     *  1.  the provided priority is {@code >} {@link #namePriority}
+     *  2.  the provided priority is {@code ==} {@link #namePriority} AND `overrideIfSamePriority` is `true`
+     * Otherwise, returns {@code null}
+     *
+     * @param namePriority the priority for the name. See also the {@code AbstractSpan#PRIO_*} constants.
+     * @return the name {@link StringBuilder} if the provided priority is {@code >=} {@link #namePriority}, {@code null} otherwise.
```

Comment:
Note to self - adjust this comment and add description for the second parameter (copy + pasta)